tb_pulumi/ec2.py

```diff
@@ -0,0 +1,356 @@
+                        'protocol': 'tcp',
+                        'from_port': target_port,
+                        'to_port': target_port,
+                    }
+                ],
+            },
+            tags=self.tags,
+            opts=pulumi.ResourceOptions(parent=self),
+        ).resources
+
+        # Build the load balancer first, as other resources must be attached to it later
+        self.resources['nlb'] = aws.alb.LoadBalancer(
+            f'{name}-nlb',
+            enable_cross_zone_load_balancing=True,
+            internal=internal,
+            load_balancer_type='network',
+            name=name,
+            security_groups=[self.resources['security_group_with_rules']['sg']],
+            subnets=[subnet.id for subnet in subnets],
+            tags=self.tags,
+            opts=pulumi.ResourceOptions(parent=self, depends_on=[self.resources['security_group_with_rules']['sg']]),
+            **kwargs,
+        )
+
+        # Build and attach a target group
+        self.resources['target_group'] = aws.lb.TargetGroup(
+            f'{name}-targetgroup',
+            health_check={
+                'enabled': True,
+                'healthy_threshold': 3,
+                'interval': 20,
+                'port': target_port,
+                'protocol': 'TCP',
+                'timeout': 10,
+                'unhealthy_threshold': 3,
+            },
+            load_balancing_cross_zone_enabled=True,
+            name=name,
+            port=target_port,
+            protocol='TCP',
+            target_type='ip',
+            vpc_id=subnets[0].vpc_id,
+            tags=self.tags,
+            opts=pulumi.ResourceOptions(parent=self, depends_on=[self.resources['nlb']]),
+        )
+
+        # Add targets to the target group
+        self.resources['target_group_attachments'] = []
+        for idx, ip in enumerate(ips):
+            self.resources['target_group_attachments'].append(
+                aws.lb.TargetGroupAttachment(
+                    f'{name}-tga-{idx}',
+                    target_group_arn=self.resources['target_group'].arn,
+                    target_id=ip,
+                    port=target_port,
+                    opts=pulumi.ResourceOptions(parent=self, depends_on=[self.resources['target_group']]),
+                )
+            )
+
+        # Build the listener, sending traffic to the target group
+        self.resources['listener'] = aws.lb.Listener(
+            f'{name}-listener',
+            default_actions=[{'type': 'forward', 'target_group_arn': self.resources['target_group'].arn}],
+            load_balancer_arn=self.resources['nlb'].arn,
+            port=listener_port,
+            protocol='TCP',
+            tags=self.tags,
+            opts=pulumi.ResourceOptions(
+                parent=self, depends_on=[self.resources['nlb'], self.resources['target_group']]
+            ),
+        )
+
+        self.finish()
+
+
+class SshableInstance(tb_pulumi.ThunderbirdComponentResource):
```

Comment:
Gosh there's really no good way to PascalCase this huh.

Reply:
No, it looks ugly no matter how you do it. I opted for this because the AWS API uses this convention of lowercasing initialisms.